### PR TITLE
feat(macos): Add `unhide_application` method, closes #182

### LINKED ISSUE #182: feat: `app.hide()` and `app.show()`
Would be nice to have the ability to hide the app, so that the app loses focus, just like the `MyApp > Hide MyApp` menu item. Mostly useful for macOS

### PR DESCRIPTION
**Pull request by probablykasper**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Closes #182.

### CODE REVIEW

**Comment by probablykasper** [comment]
For reference, unhide is documented here: https://developer.apple.com/documentation/appkit/nsapplication/1428761-unhide

It doesn't activate the app (even though the documentation made me think it should)

**Review by wusyong — APPROVED** [review]

File: src/platform/macos.rs [hunk 1]
```diff
@@ -443,13 +443,21 @@ impl MonitorHandleExtMacOS for MonitorHandle {
 
 /// Additional methods on `EventLoopWindowTarget` that are specific to macOS.
 pub trait EventLoopWindowTargetExtMacOS {
+  /// Unhide the application.
+  fn unhide_application(&self);
```
Comment (amrbashir) [inline src/platform/macos.rs]: can we move this down after `hide_application()`
```suggestion
  /// Show the entire application, see [`EventLoopWindowTargetExtMacOs::hide_application()`].
  fn show_application(&self);
```
 Also don't forget to update the changelog.

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/platform/macos.rs (hunk 1 above)
Comment (kasper9n) [reply src/platform/macos.rs]: Sure

**Review by amrbashir — APPROVED** [review]

File: src/platform/macos.rs (hunk 1 above)
Comment (kasper9n) [reply src/platform/macos.rs]: Couldn't get that doc link to work so just removed it

Comment (amrbashir) [reply src/platform/macos.rs]: Could you try to fix it ? I think it should be ``[`EventLoopWindowTargetExtMacOs::hide_application`]`` or ``[crate::platform::macos::EventLoopWindowTargetExtMacOs::hide_application]``

Comment (kasper9n) [reply src/platform/macos.rs]: Tried both of those 😕

Comment (amrbashir) [reply src/platform/macos.rs]: cc @wusyong

Comment (wusyong) [reply src/platform/macos.rs]: If that doesn't work, just use old school markdown link syntax. It's getting too long anyway

Comment (amrbashir) [reply src/platform/macos.rs]: so ``[`hide_application`](crate::platform::macos::EventLoopWindowTargetExtMacOs::hide_application)`` ?

Comment (kasper9n) [reply src/platform/macos.rs]: Rustdoc documentation says this about linking, maybe that's the issue:
> Associated items (functions, types, and constants) are supported, but not for blanket trait implementations.

Comment (amrbashir) [reply src/platform/macos.rs]: if it doesn't work, lets move on, it is not necessary.

**Pull request merged**